\documentclass[10pt]{article}
\usepackage[left=1in,top=1in,right=1in,bottom=1in,nohead]{geometry}

%PloS packages
\usepackage{cite}
\usepackage{color} 
%comment out for single
\usepackage{setspace} 
\doublespacing
\topmargin 0.0cm
\oddsidemargin 0.5cm
\evensidemargin 0.5cm
\textwidth 16cm 
\textheight 21cm
\usepackage[labelfont=bf,labelsep=period,justification=raggedright]{caption}
\bibliographystyle{plos2009}
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother
\date{}
\pagestyle{myheadings}

\usepackage{geometry}                % See geometry.pdf to learn the layout options. There are lots.
\geometry{a4paper}                   % ... or a4paper or a5paper or ... 
\usepackage{wrapfig}	%in-line figures
\usepackage[square, comma, sort&compress]{natbib}	%bibliography
\usepackage{pslatex} 	%for times new roman
\usepackage[parfill]{parskip}    % Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}
\usepackage{amssymb}
\usepackage{epstopdf}
\usepackage{booktabs}
\usepackage{amsmath}
\usepackage{setspace}
\usepackage{lscape} % add landscapes
\usepackage{color} % add colored text

\DeclareGraphicsRule{.tif}{png}{.png}{`convert #1 `dirname #1`/`basename #1 .tif`.png}

\begin{document}



\thispagestyle{empty}
\pagestyle{empty}


\begin{flushleft}
{\Large
\textbf{Genetic Basis of Metabolome Variation in Yeast}
}


J.S. Breunig, Lewis-Sigler Institute for Integrative Genomics \& Department of Molecular Biology, Princeton University, Princeton, NJ, USA

S.R. Hacket, Lewis-Sigler Institute for Integrative Genomics \& Graduate Program in Quantitative and Computational Biology, Princeton University, Princeton, NJ, USA

J.D. Rabinowitz, Lewis-Sigler Institute for Integrative Genomics \& Department of Chemistry, Princeton University, Princeton, NJ, USA

L. Kruglyak, Lewis-Sigler Institute for Integrative Genomics, Department of Ecology and Evolutionary Biology, \& Howard Hughes Medical Institute, Princeton University, Princeton, NJ, USA

$\ast$ E-mail: kruglyak@princeton.edu

\end{flushleft}


\noindent \textbf{Metabolism, the conversion of nutrients into usable energy and biochemical building blocks, is an essential feature of all cells. The genetic factors responsible for inter- individual metabolic variability remain poorly understood. To investigate genetic causes of metabolome variation, we measured the concentrations of 74 metabolites across $\sim$ 100 segregants from an \textit{Saccharomyces cerevisiae} cross by liquid chromatography-tandem mass spectrometry. We found 52 quantitative trait loci for 34 metabolites. These included linkages due to overt changes in metabolic genes, e.g., linking pyrimidine intermediates to the deletion of \textit{ura3}. They also included linkages not directly related to metabolic enzymes, such as those for five central carbon metabolites to \textit{ira2}, a Ras/PKA pathway regulator, and for the metabolites, S-adenosyl-methionine and S-adenosyl-homocysteine to \textit{slt2}, a MAP kinase involved in cell wall integrity. The variant of \textit{ira2} that elevates metabolite levels also increases glycolytic flux. These results highlight the potential to achieve diverse metabolic phenotypes through both direct modulation of metabolic enzymes and changes in global cellular regulators.}

\noindent \textbf{Author Summary}

Many traits, from human height to \textit{E. coli} growth rate, quantitatively vary across members of a species. Among the most medically and agriculturally important traits are levels of cellular metabolites, such as cholesterol levels in humans or starch in food crops. Metabolic variation in yeast also holds practical importance with some \textit{Saccharomyces} strains better suited to making ethanol for biofuel and others to making flavourful wine. Metabolic variation across yeast can be used to gain insight into the general principles regulating metabolite abundancee in eukaryotes. To this end, we have examined inter-strain differences in metabolism in over 100 closely related \textit{S. cerevisiae} strains. We identified over 50 genetic loci that control the levels of specific metabolites. These include not only loci encoding metabolic enzymes, but also those encoding global cellular regulators. For example, differences in the sequence of \textit{ira2}, an inhibitor of Ras, lead to differences in central carbon metabolite levels, and polymorphisms in \textit{slt2}, a poorly characterized MAP kinase, specifically alter levels of sulfur-containing metabolites. These findings provide insights into the mechanisms cells use to control the levels of critical metabolites such as citrate and S-adenosyl-methionine, and highlight the ability of polymorphisms in non-metabolic proteins to control metabolite concentrations.

Inter-individual differences in metabolism are of substantial biological importance. In humans, they underlie susceptibility to type II diabetes \cite{Bijlsma:2006wx}, obesity \cite{Sladek:2007gs} and Crohn’s disease \cite{Marchesi:2007gz}, while in yeast they contribute to the flavor profile of wine and to the efficiency of bioethanol generation \cite{Keasling:2008hb, Wahlbom:2003uc}. Accordingly, there has been growing interest in identifying the genetic loci responsible for inter-individual metabolome differences.

The relationship between the metabolome and the genome has been increasingly studied over the past decade, most thoroughly in the plant community \cite{Fiehn:2002vc, Goodacre:2004dg, Hall:2006hp, Schauer:2006ei}. Initial investigations followed metabolomic alterations in response to gene knockouts \cite{Goodacre:2004dg, Allen:2003id, Benfey:2008gd, VillasBoas:2005jq, Smedsgaard:2005dn}, an approach that proved valuable for annotating gene functions \cite{Clasquin:2011ev}. Recently, there has been increased interest in decoding metabolic variation due to natural perturbations using quantitative genetics \cite{DellaPenna:2008dp}. Quantitative trait locus (QTL) studies have been performed on enzyme activities and metabolite concentrations in plants, with greatest success for secondary metabolites \cite{McMullen:1998vc, Bost:1999ux, MitchellOlds:1998us, Keurentjes:2006ik, Lisec:2008bs, Bentsink:2000tg, Hobbs:2004hw, Kliebenstein:2001wm, Wentzell:2007dr}. Association of metabolite abundance variation with unsuspected genetic determinants has demonstrated the potential of metabolite QTL (mQTL) analysis for both identifying genes with previously unknown enzymatic roles \cite{McMullen:1998vc}, as well as revealing novel reactions \cite{Keurentjes:2006ik}.

These methods have been scaled to global metabolomic analysis; determining how levels of metabolites are associated with segregation across intercrosses of mice, \textit{A. thaliana} and yeast \cite{Dumas:2007cj, Rowe:2008ty, Zhu:2012cd}.  This has demonstrated that there is substantial genetic variation in primary and secondary metabolites, and this variation is governed by the segregation of relatively few mQTL hotspots \cite{Rowe:2008ty, Zhu:2012cd} whose epistatic interaction further shapes the metabolome \cite{Rowe:2008ty}.  These mQTL hotspots generally coincide with known eQTL hotspots, highlighting the extensive pleiotropy of these regions.  While these studies have been able to associate regions of the genome with metabolic alterations, the residual unexplained heritability of these studies can be extensive, raising important questions about the power and reproducibility of QTL and mQTL analysis.  Furthermore, the resolution of 100-200 F2 intercrosses is limited and identifying genetic associations has typically entailed identifying a locus of interest and reported on proximity to pathway-related enzymes, without searching rigorously for other linked genes that might play a regulatory role.    

With the goal of discovering potential novel regulators of primary metabolism, we examined 74 metabolites involved in highly conserved core metabolic pathways of central carbon metabolism and nucleotide and amino acid biosynthesis. We found 52 significant linkages, and experimentally verified the genes underlying three major linkage hotspots, including two linked genes responsible for altering S-adenosyl-methionine levels, neither with known metabolic roles. Additionally, we compared our metabolite results with the expression QTL results for the same cross \cite{Smith:2008vy}, and found six overlapping hotspots. The largest mQTL hotspot is shared with the largest hotspot in the transcript data, and is due to polymorphisms in a global regulator of cell signaling, \textit{ira2}. Interestingly, while the expression QTLs linked to \textit{ira2} were enriched for central metabolic enzymes, the variant of \textit{ira2} that promoted high metabolite concentrations favored low enzyme transcript levels. This dichotomy is explained by \textit{ira2}-linked transcripts being primarily involved in oxidative metabolism, while the linked metabolites are mainly involved in fermentation. These findings highlight the ability of mQTL analysis to reveal metabolic regulatory mechanisms.

\vspace{10mm}
\begin{center}
\textbf{RESULTS}
\end{center}
\vspace{10mm}

To identify genetic loci responsible for inter-individual differences in the metabolome, we used a well-studied cross between a laboratory strain of yeast, BY4716, and a vineyard isolate, RM11-1a (hereafter, BY and RM, respectively). These strains have both been sequenced, and they differ at $\sim$0.6\% of base pairs \cite{Foss:2007ej}. Over 100 segregants from the cross have been densely genotyped and used in studies of the genetic basis of variation in protein and transcript levels \cite{Smith:2008vy, Foss:2007ej, Brem:2005cn, Brem:2005gh} and a number of other phenotypes \cite{Perlstein:2007ku, Perlstein:2006ds}.

Intracellular metabolites were harvested from yeast growing exponentially on aerobic, glucose-containing minimal medium by direct quenching and extraction in cold organic solvent \cite{Brauer:2006dt}. The samples were then analyzed using two complementary targeted LC-MS/MS methods, one in positive ion mode and the other in negative ion mode \cite{Lu:2008fj}. Each method provides three-fold confirmation of metabolite identity based on parent ion mass, gas-phase fragmentation to a characteristic daughter ion, and LC retention time match to authenticated metabolite standard. We collected measurements from 13 independent replicates of the BY strain, 18 independent replicates of the RM strain, and two independent samples from each of 114 segregants. 105 compounds were reliably detected in at least one parent strain, and 79 of these were significantly different between the two strains at a false discovery rate (FDR) of 5\%. 74 of the 105 known compounds were measured in at least one-quarter of the segregants, and these 74 compounds were used for linkage analysis.  No compounds found to have significant mQTLs were as abundant in mock extracted samples as biological samples.

Many of these compounds showed inheritance patterns consistent with a complex underlying genetic basis. Based on the methods described for transcripts in Brem et. al. 2005 \cite{Brem:2005gh}, we determined that 14 compounds showed transgressive segregation (the range in the segregants significantly exceeded that spanned by the parent strains), 28 showed directional genetics (most segregants had levels intermediate between the parent strains), and 32 showed evidence of epistatic interactions, with substantial overlap among these categories. The observation of genetic complexity for most metabolite levels is concordant with what has been observed for other traits in this cross.

\textbf{Linkage analysis}

We tested for linkage with R/qtl \cite{Broman:2003wq}, and used permutations to establish that a LOD score of 3.4 corresponds to an empirical FDR of 10\%. Of the 74 compounds tested, 34 showed at least one significant linkage (metabolite quantitative trait locus or mQTL; Table S1). The majority of these compounds (21 of 34) had one mQTL, 9 had two mQTLs, three had three mQTLs and one had four mQTLs, for a total of 52 detected mQTL. Almost all the compounds for which mQTL were detected differed significantly between the parents strains at an FDR of 5\% (29 of 34).  For 24 compounds that differed significantly between the parents strains, we did not detect mQTL, most likely due to complex underlying genetics with multiple loci of small effect. The mQTL were not uniformly distributed along the genome; rather, most fell within 8 “hot spots” with 3 or more compounds linking to each (Figures \ref{F1} and \ref{F2}, Materials and Methods). The observation of such hot spots, previously seen for other classes of traits, implies the presence of underlying polymorphisms with broad effects on the metabolome. These hot spots are discussed further below.


\begin{figure}[p]
\includegraphics[width = 9in, angle = 90]{Figures/Figure1-MetaboliteLinkages.pdf}
\caption{Distribution of significant linkages across the genome. Metabolite linkages that exceeded the 0.1 significance threshold are plotted based on their most significant marker’s genome location with a 95\% confidence interval. Continuous vertical lines represent chromosome ends. Numerals are placed at chromosomes’ center. Genes investigated in this study are shown at top.  All unknown metabolite QTLs are combined into a single class.}
\label{F1}
\end{figure}

\begin{figure}[p]
\includegraphics[width = 6in]{Figures/Figure2-MetabolitesVsTranscripts.pdf}
\caption{Similarities between metabolite and transcript linkage distributions. The number of significant linkages are binned in 10 kb increments, the count for these bins are plotted. Linkage distributions for transcripts are shown at top, metabolites at bottom. Dotted blue lines show chromosome ends. Red lines show the hotspot cutoff (see Methods for calculation).}
\label{F2}
\end{figure}


\textbf{Transcriptome and metabolome variation}

We compared the metabolite linkage results with those for transcript abundance in the same cross \cite{Smith:2008vy}. Transcript linkages also clustered in hot spots, and the hot spots for metabolites and transcripts show a significant overlap in location, with six of eight metabolite hot sports corresponding also to transcript hot spots (p $<$ 0.0001, based on permutation test) (Figure \ref{F2}). The hot spots unique to metabolism included hot spot m8 on chromosome XVI (linked to levels of ribose-phosphate, aspartate and glutamate) and hot spot m5 on chromosome VIII (linked to levels of S-adenosyl-homocysteine, S-adenosyl-methionine, and thiamine). This latter hot spot is especially interesting since regulation of the methionine cycle is poorly understood in eukaryotes, despite being implicated in cardiovascular disease \cite{Kampfer:2010bl, Fetrow:2001ty}. It is discussed in depth below.

\textbf{Metabolic genes in confidence intervals}

To determine whether changes in metabolites tend to be linked to genes with known roles in metabolism, we carried out functional enrichment analysis of genes located in mQTL confidence intervals. Genes were classified as ‘metabolic’ based on inclusion in the iMM904 metabolism model \cite{Mo:2009fk}. The mQTL confidence intervals were found to be significantly enriched for metabolic genes. 471 out of a total of 904 metabolic genes in the yeast genome partially or completely overlapped with an mQTL 95\% confidence interval. In random permutations, 191 to 512 metabolic genes overlapped with genomic intervals of the same size, and the number observed for the actual confidence intervals (471) fell in the top 0.07\% of the permuted distribution (Figure S1). Each mQTL confidence interval was also examined specifically for the presence of metabolic genes in the same pathway as the linked metabolite (Table S2). Over half (31/52) of the confidence intervals were found to contain at least one metabolic gene from one of the pathways involving the linked metabolite.

\textbf{The \textit{ura3} hot spot}

Levels of five metabolites linked to a hot spot on chromosome V: orotate, orotidine, orotidine-5’-phosphate, UDP-D-glucose, and UDP-N-acetyl-glucosamine. All five are intermediates or products of pyrimidine biosynthesis (Figure \ref{F3}). \textit{Ura3}, a pyrimidine biosynthesis gene which carries an engineered deletion in the RM strain, is contained within the hot spot and lies within the 95\% mQTL confidence intervals for all five compounds (Figure S2). Compounds upstream of \textit{ura3} in the pathway show the greatest differences in abundance (as much as 128-fold), and particularly strong linkages (Figure \ref{F3}). 

\begin{figure}[p]
\includegraphics[width = 6in]{Figures/Figure3-URA3Pathway.pdf}
\caption{Levels of pyrimidine intermediates and products differ based on the \textit{ura3} allele inherited. The relevant portions of the pathway are shown, with measured metabolites in red. The location of \textit{ura3} in the pathway is shown in green. The accompanying plots show phenotype distribution of the segregants based only on the allele of \textit{ura3} inherited: RM in purple, BY in orange. The \textit{ura3} gene is defective in RM. All metabolite levels are log2(Segregant / RM). Compounds (LOD scores) that were significantly linked to \textit{ura3} locus are bolded.}
\label{F3}
\end{figure}

To confirm that this mQTL hotspot was governed by segregation of the engineered \textit{ura3} deletion, \textit{ura3}$\bigtriangleup$, this RM allele was inserted into a BY background and compounds differing between BY and BY\textit{ura3}$\bigtriangleup$ were assessed.  Using a two-tailed t-test, four compounds were found to differ between these two conditions at a 0.05 FDR \cite{Storey:2003cj}.  The deletion resulted in the elevation of all four compounds, including orotate and orotidine-5'phosphate, two of the 5 compounds that were associated with this mQTL hotspot.  The other three compounds that were significantly associated by mQTL were found to only weakly differ or were not ascertained in this confirmatory analysis, suggesting that we have less power to detect their divergence, and their magnitude is weaker.  The remaining two compounds that were elevated in BY\textit{ura3}$\bigtriangleup$, N-carbamoyl-aspartate and inosine diphosphate (IDP) were identified subsequent to the quantification of segregants, so their global linkage could not be assessed, however their association to this locus is intriguing.  N-carbamoyl-aspartate is the product of the first pyrimidine anabolic reaction and is the precursor of dihydroorotate, so its accumulation is not surprising.  In contrast, because IDP is a purine, its massive accumulation under pyrimidine knock-out, far greater than other purine metabolites, is unexpected and suggests a purpose.  Inosine, whose abundance is not associated with \textit{ura3}$\bigtriangleup$, has previously been shown to mediate pyrimidine scavenging \cite{Tozzi:1981ga} while IMP is a known allosteric regulator of carbamoyl phosphate synthetase \cite{Trotta:1974ws}.  IDP's massive accumulation suggest an abundant pool of IMP driving flux into pyrimidine metabolism, or it may be working directly as a secondary mediator of one of these functions.  These results demonstrate that our approach can link changes in metabolite levels to a polymorphism (in this case, an engineered one) in a gene known to participate in the biosynthesis of the relevant metabolites.



\textbf{\textit{Slt2} and \textit{erc1} polymorphisms impact S-adenosyl-methionine levels}

The mQTL hot spot on chromosome VIII (m5) is linked to levels of 3 metabolites: thiamine, S-adenosyl-methionine (SAM), and S-adenosyl-homocysteine (SAH) (Table S1). The overlap among the 95\% confidence intervals of the mQTL for each compound covers a region containing all or part of 14 genes (Figure S3). None of the genes in this region have a known connection with the sulfur-assimilation pathway. We identified \textit{slt2} as a candidate for further evaluation due to the presence of a two amino acid indel polymorphism between BY and RM in a polyglutamine track; variation in the number of glutamines in this track has previously been implicated in stress response \cite{deLlanos:2010jt}.

Segregants inheriting the RM allele of \textit{slt2} had significantly higher levels of SAM and SAH (Figure \ref{F4}). To test the allelic effect of \textit{slt2}, we created allele-replacement strains in both parental backgrounds and compared metabolite levels to those in the parent strains (Figure \ref{F5}). In the BY background, the RM allele of \textit{slt2} did not raise SAH levels above the limit of detection, nor did it result in a significant change for SAM (p = 0.1598). However, in the RM background, the BY allele of \textit{slt2} resulted in a three-fold decrease for both SAM and SAH (Figure \ref{F5}; p $<$ 0.001). The difference in the effects of the allele swaps in the two backgrounds implies an interaction between the allelic status of \textit{slt2} and other loci.

We considered the possibility that the effect of this locus is due to polymorphisms in multiple linked genes. We investigated a nearby gene, \textit{erc1}, due to the presence of an indel polymorphism that causes a frameshift which alters 37 residues and extends the peptide by 43 amino acids in the RM background. \textit{Erc1} has also been shown to have an effect on SAM levels when overexpressed in \'{s}ake strains of S. cerevisiae \cite{Choi:2009vg, Lee:2010ta, Shiomi:1991vo}. \textit{Erc1} is located 3kb (approximately 1 cM) from \textit{slt2}, and thus the alleles of the two genes segregate together as a haplotype. We used the \textit{slt2} allele replacement strains to create strains in which both genes were replaced with the alternative alleles. In the BY background, replacing both \textit{slt2} and \textit{erc1} with the RM alleles led to a significant increase in SAM (p = 0.019) compared to the original BY strain, but the level of SAM was still much lower than in RM (Figure \ref{F5}). In the RM background, replacing both genes with the BY alleles led to significantly lower levels of both metabolites compared to either the original RM strain or to the \textit{slt2} replacement alone (p $<$ 0.001 for all comparisons). These results suggest that polymorphisms in both \textit{slt2} and \textit{erc1} alter the levels of SAM-cycle compounds in these strains, but that other undetected loci also play a role in the observed variation.

\begin{figure}[p]
\includegraphics[width = 6in]{Figures/Figure4-SLT2Impact.pdf}
\caption{Levels of sulfur-assimilation intermediates differ based on the \textit{slt2} allele inherited. The relevant portions of the pathway are shown, with measured metabolites in red. The accompanying plots show phenotype distribution of the segregants based only on the allele of \textit{slt2} inherited: RM in purple, BY in orange. All metabolite levels are log2(Segregant / RM). Compounds (LOD scores) that were significantly linked to \textit{slt2} locus are bolded.}
\label{F4}
\end{figure}

\begin{figure}[p]
\includegraphics[width = 6in]{Figures/Figure5-slt2anderc1effects.pdf}
\caption{RM-inheriting segregants for \textit{slt2} and \textit{erc1} show significantly higher levels for SAM.  Intensities (mean $\pm$ standard error) of SAM are plotted based upon the allele of \textit{slt2} (top) and \textit{slt2} and \textit{erc1} (bottom). Absolute intensities for BY background (diamonds) and RM background (squares) are plotted on the left axis while segregants’ (triangles) relative intensities are plotted on the right axis.}
\label{F5}
\end{figure}

\textbf{IRA2 polymorphisms alter central metabolites}

A mQTL hot spot on chromosome XV (m6) is linked to five central carbon metabolites: glucose-6-phosphate (G6P) and its isomers (which were not distinguished by the LC-MS method used here), fructose-1,6-bisphosphate (FBP), sedoheptulose 7-phosphate (S7P), dihydroxyacetone phosphate (DHAP), and (iso)citrate. The overlap among the 95\% confidence intervals of the mQTL for each compound covers a region containing all or part of 13 genes (Figure S4). We focused on \textit{ira2} as a candidate gene due to its known function as a regulator of the Ras/PKA pathway \cite{Tanaka:1991ts} and because we previously showed that polymorphisms in \textit{ira2} underlie a major eQTL hot spot (t16) at the same locus in this cross \cite{Smith:2008vy, Storey:2005hm}. \textit{Ira2} is a Ras-related GTPase \cite{Tanaka:1991ts, Parrini:1995tr, Broach:1991ur}, with \textit{ira2}-catalyzed GTP hydrolysis leading to inactivation of Ras. The eQTL expression patterns suggested that \textit{ira2} is hypoactive in the BY strain.

Segregants inheriting the BY allele of \textit{ira2} showed higher levels of all five linked metabolites compared to those inheriting the RM allele (Figure \ref{F6}). To test the allelic effect of \textit{ira2}, we compared metabolite levels of \textit{ira2} allele-replacement strains in both backgrounds \cite{Smith:2008vy} to the original parental strains (for FBP, see Figure \ref{F7}; for other metabolites, see Figure S5). In the RM background, the BY allele of \textit{ira2} led to significantly higher levels of 3 compounds (p $<$ 0.01 for sedoheptulose-7-phosphate, FBP, DHAP). In the BY background, the RM allele of \textit{ira2} led to significantly lower levels of all 5 metabolites (p $<$ 0.05). These results demonstrate that polymorphisms in \textit{ira2} contribute to the observed variation in these five central metabolites.

Metabolites can accumulate due to either increased production or decreased consumption. To distinguish whether the BY allele of \textit{ira2} was enhancing central carbon metabolic flux versus inhibiting central metabolite consumption, we analyzed glucose uptake in the BY and RM parental strains, as well as in \textit{ira2} allele-replacement strains in both backgrounds. Glucose uptake rate did not different significantly between the two parental strains. In the two allele-replacement strains, however, glucose uptake markedly diverged. In the RM background, the BY allele of \textit{ira2} led to 45\% faster glucose uptake, whereas in the BY background, the RM allele led to a 20\% decrease (Figure \ref{F7}).  To determine if this change in glucose uptake translates into glycolytic flux, ethanol excretion of these \textit{ira2} allele-swap strains was quantified using NMR and confirmed that the BY allele significantly increases ethanol excretion (p $<$ 0.05).  These results demonstrate that polymorphisms in \textit{ira2} control central carbon metabolic flux, with the BY allele inducing both higher metabolite levels and fluxes. In the parental strains, the metabolic flux impact of the \textit{ira2} polymorphism is presumably offset by differences at other loci.

As polymorphisms in \textit{ira2} result in differences in expression of $\sim$1300 genes \cite{Smith:2008vy}, we considered whether expression differences in central carbon metabolism genes might underlie the observed metabolic changes. Of 70 known central carbon metabolism genes (i.e., those with roles in glycolysis, pentose phosphate pathway, citric acid cycle, and oxidative phosphorylation from yeastgenome.org), expression of 32 linked to the \textit{ira2} locus in glucose media (Table S3).  This significantly exceeds the number of linkages expected for a random set of genes (p $<$ 0.01, Fischer’s exact test). Remarkably, of the 32 linked genes, 28 are less highly expressed in the BY strain, which has higher levels of G6P, FBP, S7P, DHAP, and (iso)citrate. Thus, paradoxically, the BY allele of \textit{ira2} promotes higher central carbon metabolite levels while repressing central carbon metabolism gene expression.

Insight into this paradox is provided by the nature of the regulated genes: 28 of the 32 central carbon metabolism genes linking to \textit{ira2} tend to be more highly expressed in ethanol than in glucose \cite{Smith:2008vy}; i.e., the primary transcriptional regulatory role of \textit{ira2} seems to be in enhancing expression of genes required for respiratory growth. In contrast, the linked metabolites are indicative, with the exception of (iso)citrate, of active fermentation. The accumulation of (iso)citrate in the BY strain is consistent with the BY allele of \textit{ira2} being associated with lower expression of the primary isocitrate consuming enzyme \textit{idh1}. Taken together with the data showing that the BY allele of \textit{ira2} promotes glucose fermentation, one obtains a coherent view: \textit{ira2} activity is lower in the BY strain. This leads to decreased expression of genes required for respiration, more need for fermentative ATP production, and higher levels of the glycolytic intermediates G6P, FBP, and DHAP.

\begin{figure}[p]
\includegraphics[width = 6in]{Figures/Figure6-IRA2Impact.pdf}
\caption{Levels of glycolysis, pentose phosphate pathway and TCA intermediates differ based on the \textit{ira2} allele inherited. The relevant portions of the pathway are shown, with measured metabolites in red and significant linkages bolded. The accompanying plots show phenotype distribution of the segregants based only on the allele of IRA2 inherited: RM in purple, BY in orange. All metabolite levels are log2(Segregant / RM). LOD score for the closest marker is also shown. *includes analytically indistinguishable isomers.}
\label{F6}
\end{figure}

\begin{figure}[p]
\includegraphics[width = 6in]{Figures/Figure7-ira2effects.pdf}
\caption{RM-inheriting segregants for \textit{ira2} show significantly lower levels for fructose-1,6-bisphosphate. Intensities (mean $\pm$ standard error) of FBP are plotted based upon the allele of \textit{ira2}.  Absolute intensities for BY background (diamonds) and RM background (squares) are plotted on the left axis while segregants’ (triangles) relative intensities are plotted on the right axis.}
\label{F7}
\end{figure}

\textbf{Heritability and mQTL reproducibility}

We can only relate metabolite abundance variation to genetic heterogeneity across segregants when there is substantial genetic variation affecting metabolite levels in the first place.  Previous estimates of broad-sense heritability \cite{Lynch:1998vx} in \textit{A. thaliana} have suggested moderate heritability of metabolite traits across globally-distributed strains \cite{Keurentjes:2006ik}, while segregants showed substantially lower heritability of metabolite traits than expression traits (an average of 25\% and 65\% respectively) \cite{Rowe:2008ty, West:2006bk}.  We find extensive heritable variation of metabolite abundance in this study, with an average broad-sense heritability of 62\%.  This indicates that there are likely larger metabolic differences segregating between BY \& RM than within the Bay $\times$ Sha \textit{A. thaliana} cross.  Greater levels of heritability across metabolites are associated with an increased number of detected mQTLs (p = 0.014); this is evident from figure \ref{QTLherit} which shows that most high heritability metabolites' variation is associated with at least one locus.  The effects sizes of these QTLs can be found by determining the fraction of the variance in metabolite abundance that is explained using QTL genotypes (Figure \ref{QTLvarex}).  Effect sizes and the total fraction of heritability explained vary greatly across metabolites, with some mQTLs explaining the vast majority of genetic variation, others collectively explaining a sizable portion through the joint additive effects of multiple loci and others still, having small effect sizes.  The large fraction of unexplained metabolite abundance heritability could be owing to two factors: either we don't have the power to detect small effect sizes which collectively describe a significant fraction of heritability, or the tests utilized here which identify additive contributions to the narrow-sense heritability fail to capture the genetic complexity underlying metabolite abundance variation \cite{Rowe:2008ty, Bloom:2012wj}.  Both of these limitations could be partially overcome using a greater number of segregants, although adequate samples are difficult to phenotype using an analytical platform such as mass spectrometry.  

\begin{figure}[h!]
\centering
\includegraphics[width = 4in]{Figures/heritPlot.pdf}
\caption{Distribution of broad sense heritability (H$^{2}$) across measured metabolites.  Each circle represents a single metabolite, colored according to how many QTLs are associated with its abundance.  114 metabolites are shown: 74 known metabolites with 52 detected mQTL and 42 unknown metabolites (with known m/z, but unknown identity) associated with 20 additional mQTLs.}
\label{QTLherit}
\end{figure}

\begin{figure}[h!]
\centering
\includegraphics[width = 6in]{Figures/heritExplainedComb.pdf}
\caption{Fraction of broad-sense heritability explained by identified mQTLs.  Each stacked bar represents a single metabolite which was significantly associated to at least one locus.  The height of the bar is the broad-sense heritability of the metabolite's abundance, and the coloration partitions this heritability into unexplained heritability (gray), and the effects of each mapped QTL (colors).  Three examples are given to demonstrate the variable effect sizes observed across metabolites.  The distribution of metabolite abundances for a genotype is shown as a violin plot, and a 95\% confidence interval for the median of each genotype is reported with error bars.  This confidence interval was determined using a percentile bootstrapping method \cite{Davison:1997vn}.}
\label{QTLvarex}
\end{figure}

The BY $\times$ RM cross characterized in this study has previously been used to study metabolite abundance variation using quantitative NMR \cite{Zhu:2012cd}.  While the design of these studies are very similar, the use of LC-MS in this study and different handling procedures resulted in substantial differences in the observed mQTL hotspots allowing us to expand upon and provide an alternative explanation for the basis of some of these controlling regions.  Of the 56 metabolites reported in that paper, 27 match those quantified in this study, and of the 16 metabolites which they linked-to five regions of the genome, 12 were shared between our studies.  

Three hotspots appear to be largely shared between the two studies; those which we have associated with the \textit{ura3}, \textit{slt2}/\textit{erc1}, and \textit{ira2} loci.  The \textit{ura3} auxotrophy was implicated through its association with orotate and dihydro orotate elevation; metabolic effects that are confirmed, expanded upon and defi
nitively linked to \textit{ura3} in this study.  SAM and SAH were linked to the same locus as \textit{slt2}/\textit{erc1}, but mention of this mQTL hotspot or any proximal genes was neglected.  Their third mQTL hotspot mapped to the vicinity of \textit{ira2} and \textit{pmh7} and governed the abundance of glycerol, lysine, tyrosine and trehalose.  The authors concluded that variation in \textit{pmh7} was the causal source of these metabolic alterations, however the phenotype of the knockout was weak.  Because lysine was the only metabolite associated with this locus which we are able to detect in this study (and no associations were found for it) it is difficult to determine whether \textit{ira2} and \textit{phm7} work as a complex locus (similar to \textit{slt2/erc1}) or whether \textit{ira2} is this regions sole metabolic effector, with additional governance of trehalose and glycerol levels. 

The remaining mQTL hotspots of Zhu et al. were associated with amino acid metabolism and were not apparent in this work, presumably because the alterations of these metabolites are a G $\times$ E consequence of differences in growth conditions i.e. synthetic compete media in Zhu et al. 2012 and supplemented minimal media in this study.  These mQTL hotspots which depend upon growth conditions (gxemQTL), are analogous to gene-environment interaction eQTLs (gxeQTL) previously identified in this BY $\times$ RM cross \cite{Smith:2008vy}.  The identification of loci such as these whose metabolic impact is only seen under a subset of conditions could be an important method for discovering novel metabolic regulatory mechanisms.  In contrast to the transience of these unconfirmed mQTL hotspots, the permanence of the three other mQTL hotspots suggests that the causal regulators which we have found are likely to shape metabolism regardless of the nutrient condition of the cell.  Additional mQTLs found in this study (Figure \ref{F1}) may prove to control solitary metabolites or may be indicators of additional mQTL hotspots with small effects across metabolism on observed and un-ascertained metabolites. 





\clearpage
\begin{center}
\textbf{DISCUSSION}
\end{center}
\vspace{10mm}

We have used high-throughput metabolite phenotyping in a cross of two divergent strains of yeast to find 52 linkages for 34 metabolites. We detected linkages for a majority of compounds with significant differences between parental strains, as well as for a few compounds without such differences. Many metabolites show transgressive segregation, with levels in progeny strains outside the range of the parents; the parent strains likely carry alleles with opposing effects, with some segregants inheriting combinations of alleles that result in extreme metabolite levels, as also observed for transcript levels \cite{Brem:2005gh}. Such opposing effects in the parent strains were also evident in control of glycolytic flux, which is similar in the parents but diverges upon an \textit{ira2} allele swap.

\textit{Ira2} is a regulator of cell signaling, not metabolism per se. Nevertheless, allelic differences in \textit{ira2} have a broad impact on central metabolism, with the hypoactive \textit{ira2} variant found in the BY strain associated with lower levels of transcripts involved in oxidative metabolism; higher levels of citrate, glycolytic intermediates, and sedoheptulose-7-phosphate; and higher glycolytic flux. These observations are consistent with active \textit{ira2} inducing oxidative metabolic genes, which in turn decrease the glycolytic flux required to fulfill ATP production. This raises the intriguing possibility that, due to the efficiency of oxidative ATP production, the extent of “residual” oxidative phosphorylation during yeast fermentative growth is a major determinant of glycolytic flux. More direct inhibition of glycolysis by the BY variant of \textit{ira2}, e.g., through inhibition of phosphofructokinase-2, is also possible.

Perhaps the most exciting use of yeast mQTL mapping is to discover novel regulators of metabolism. In this respect, we have found linkages between levels of SAM and SAH and two proteins with no previously known metabolic regulatory role, \textit{slt2} and \textit{erc1}, which interestingly segregate as a complex haplotype. SAM and SAH are key metabolites from the perspective of epigenetics: they are substrates and products, respectively, in DNA and histone methylation. Through epigenetics or other mechanisms, they may impact a broad range of diseases, e.g., of the cardiovascular system \cite{Kampfer:2010bl, Fetrow:2001ty}, liver \cite{Mato:2002kk}, or brain \cite{Mischoulon:2002vu, Reme:2008uk, Fuso:2005wc}. \textit{Slt2} is part of a MAP kinase cascade responsible for maintaining cell wall integrity, and thus contributing to fitness during osmotic stress.  \textit{Erc1} was identified for conferring ethionine resistance \cite{deLlanos:2010jt, Choi:2009vg, Lee:2010ta, Shiomi:1991vo, MartinYken:2003wt, Chen:2005co, Lesage:2006ew}. While SAM and SAH (as well as a thiamine, which also links to the same locus), are notable for containing sulfur, neither \textit{slt2} nor \textit{erc1} is regulated transcriptionally in response to sulfur availability \cite{Boer:2003fi, Cavalieri:2000hw}. Both sulfur metabolites and \textit{slt2} have been associated with the cell cycle (in the case of \textit{slt2} via the cell cycle transcription factors \textit{swi4} and \textit{swi6}) \cite{Madden:1997ta, Blank:2009jt, Tu:2005cv, Tu:2006cl, Tu:2007cf}. The molecular mechanism by which \textit{slt2} and \textit{erc1} polymorphisms regulate SAM and SAH levels remain, however, to be elucidated.  The discovery of the underlying mechanisms may in turn inform the overall interplay between metabolism, epigenetics, and the cycle cell. Thus, mQTL analysis provides a powerful tool for integrative systems biology.

\vspace{10mm}
\begin{center}
\textbf{Materials and Methods}
\end{center}
\vspace{10mm}

\noindent \textit{Culture conditions:} We used strains generated from the cross between BY4716 (MAT $\alpha$ LYS2$\triangle$) and RM11-1a (MATa \textit{leu2}$\triangle$ \textit{ura3}$\triangle$); these strains have been extensively studied for a variety of quantitative phenotypes \cite{Smith:2008vy, Foss:2007ej, Brem:2005gh, Brem:2005cn, Perlstein:2007ku, Perlstein:2006ds, Kruglyak:1995wh}. Growth medium comprised 6.7g/L Yeast Nitrogen Base (YNB) without amino acids, 2\% (w/v) glucose as the sole carbon source, and supplemented with leucine, lysine and uracil (final concentrations 100 mg/L, 30 mg/L, 20 mg/L respectively) to complement the strain auxotrophies. Yeast were grown in this medium using a filter culture technique that enables rapid sampling of metabolism without perturbation of the cultured cells \cite{Brauer:2006dt}. In brief, strains were grown aerobically in liquid minimal medium to an OD600 $\sim$ 0.1, at which point 5 mL of the culture was transferred by filtration to the surface of an 82 mm, 0.45 $\mu$m pore size nylon membrane, which was subsequently placed atop a medium-loaded agarose plate as described in Brauer et al. \cite{Brauer:2006dt}. The filter cultures were grown aerobically to mid-log phase (OD$_{600}$ in 5 mL wash = 0.2-0.6, for 3-5 hr, approximately 2-4 doublings) before metabolism quenching and metabolome extraction. All growth was at 30$^{o}$C. Cultures were grown in triplicate, with two filters used for metabolite extraction and the third filter for OD measurement.

\noindent \textit{Metabolite extraction:} The cell-loaded filter membrane was quenched by placing it cell-side down in 2 mL of acetonitrile/methanol/water (40:40:20) at -20$^{o}$C. After 15 min, residual cells were rinsed off of the filter and the $\sim$ 2 mL cell-extraction solvent mixture was centrifuged at 13,200 rpm for 5 minutes at 4$^{o}$C to generate a clear supernatant. 90$\mu$L of this clear metabolome extract was mixed with 10$\mu$L of a mixture of isotope-labeled internal standards to yield an analysis-ready sample. Samples were stored at -4$^{o}$C until analysis, which was completed within 24 h of sample generation.

\noindent \textit{Metabolome quantitation and pre-analysis:} Two different LC separations were coupled by electrospray ionization (ESI) to Thermo TSQ Quantum triple quadrupole mass spectrometers operating in multiple reaction monitoring (MRM) mode. Positive-mode ESI was coupled to hydrophilic interaction chromatography (HILIC) on an aminopropyl column; negative-mode ESI was coupled to reversed-phase chromatography with an amine-based ion pairing agent \cite{Bajad:2006bq, Luo:2007fb}.

Raw LC-MS/MS data from both runs were analysed using the MAVEN software \cite{Melamud:2010bp}. The results of this automated analysis were manually verified in all cases. Peak quantitation was based on the average of the top three points in the peak.

For linkage analysis, compounds detected in fewer than 25\% of samples were discarded; for the remaining compounds, when signal was not detectable, raw ion counts were floored to 32, which is approximately the lower limit of detection. Duplicate samples of the same strain were averaged and then divided by the associated OD at extraction to normalize for any sample-size differences.

Every day the RM11-1a strain was also run under this method. To correct for inter-day variance in raw signal intensities, log-ratios between segregant and the same-day RM values were used for each compound.

\noindent \textit{Analysis of metabolome differences between the parental strains:} For each compound’s abundance data, an ANOVA of the form phenotype ~ strain was performed in R using the aov function to compute p-values. These p-values were then false-discovery-rate corrected to assess statistical significance. Tests for mode of inheritance were conducted according to the formulae laid out in Brem \& Kruglyak \cite{Brem:2005gh}.

\noindent \textit{Mock extraction:} To determine which metabolites may appear abundant by virtue of the extraction procedure, metabolite levels from mock extracted cells were compared to the parental strains using a one-tailed t-test and six compounds were found at levels comparable to biological samples.  Four of these metabolites were included in the media as vitamins or supplements: leucine/isoleucine, nicotinate (B$_3$), pantothenate (B$_5$), and 4-Pyridoxic acid (a B$_6$ catabolite).  Two additional metabolites had elevated levels that likely resulted from systematic contamination: deoxyribose-phosphate and D-glucono-$\delta$-lactone-6-phosphate.  No QTLs were associated with any of these compounds, so their inclusion should not impact our subsequent analysis.


\noindent \textit{Segregant linkage analysis:} We used genotypes at 2,820 markers that have been previously characterized \cite{Brem:2005gh}, giving an average spacing between markers of 4.3 kb or 1.5 cM. With over 100 segregants, we would expect to see an average of more than one recombination event between adjacent marker pairs in this cross. Linkage analysis was performed using the qtl package in R \cite{Broman:2003wq}. We used the normal model and nonparametric method, assessing significance through the built-in permutation test. For every metabolite we computed 100 permutations of the qtl profile; linkage scores that were in the top 10\% of this set were considered significant. This cutoff differs for each metabolite, ranging from a LOD score of 3.14 to 3.58 with an average of 3.35. We calculated confidence intervals using the bayesint function with a probabilit y of 0.95. This is generally considered more conservative than intervals calculated based on a 1.5 LOD drop; secondary peaks on the same chromosome will result in larger intervals.

\noindent \textit{Allele replacement strains:} The allele replacement strains for IRA2, SLT2, and ERC1 were constructed according to methods laid out in Gray et al. \cite{Gray:2005jt} and Smith et al. \cite{Smith:2008vy}. The strains used were BY4724 (MATa LYS2$\triangle$ URA3$\triangle$), BY4724 IRA2$^{RM}$, BY SLT2$^{RM}$, BY SLT2$^{RM}$ ERC1$^{RM}$, ACY753 (an RM MATa URA3$\triangle$), and RM IRA2$^{BY}$, RM SLT2$^{BY}$, RM SLT2$^{BY}$ ERC1$^{BY}$. Allele swap strains were compared to their parental strain using paired t-tests.


\noindent \textit{Identification of metabolic genes in confidence intervals:} Confidence intervals for each QTL were computed as described above. Using the intervals package in R and the position and name of metabolic genes from Mo et al. \cite{Mo:2009fk}, we created a dataset of all metabolic genes in the \textit{S. cerevisiae} genome. The intervals\_overlap() command returned how many and which metabolic genes fully or partially overlap with our confidence intervals. To compute significance for all confidence intervals, we randomly permuted the position of the intervals 10,000 times, each time recording the total number of metabolic genes contained in the intervals.

To look at pathway-specific metabolic genes for each metabolite we compared the list of genes in all pathways that SGD has listed for that metabolite with the list of all metabolic genes in that metabolite’s confidence interval (pathway information was downloaded from Yeast Biochemical Database available at \textit{Saccharomyces} gene database http://www.yeastgenome.org/biocyc on 29 September 2009). For metabolites with multiple linkages, each confidence interval was examined separately.

\noindent \textit{Comparison between metabolite and transcript datasets:} All transcript data was taken from Smith and Kruglyak \cite{Smith:2008vy}, using only the data for glucose-grown cells.

For comparing linkage location, the genome was broken into 10 kb bins and the peak of each linkage (transcript and metabolite) was assigned to a bin. A bin was considered to have an excess of linkages if the number exceeded the number expected by chance by Poisson distribution. Given the number of metabolite-linkages (52) and bins (1216) we have $\lambda$ = 0.0428, and we used a Bonferroni corrected significance (p $<$ 4.11*10-5); this resulted in significance for any bin that linked to three or more metabolites. For transcript-linkages $\lambda$ = 4.151 and the significant hotspots are defined by have 14 or more linkages. Hotspots in immediately adjacent bins were considered part of the same hotspot. When considering hotspots between the datasets, they were considered shared only if they inhabited the same linkage bin.

\noindent \textit{Heritability and study reproducibility:} For each metabolite, segregants with 2 quantifiable biological replicates were isolated and the variance within replicates was compared to the total across all samples.  This is effectively subtracting the environmental variance from the total phenotypic variance to yield the genetic variance.  The ratio of genetic variance to phenotypic variance is the broad sense heritability (equation \ref{heriteq})
\large{
\begin{align}
\hat{\sigma^{2}_{s}} &= \sum_{r = 1}^{2}\left(\frac{X_{sr} - \overline{X}_{s}}{2}\right)^2 \cdot \frac{2}{2 - 1}\notag\\
H^{2} &= 1 -  \frac{\sum_{s}^{S}2\hat{\sigma^{2}_{s}}}{\sum_{s}^{S}\sum_{r = 1}^{2}(X_{sr} - \overline{X})^2}\label{heriteq}
\end{align}
}
The association between the number of QTLs found for a metabolite and the metabolite's heritability was found by modeling the number of detected QTLs as an approximately poisson trait and predicting this value using poisson regression.  


\vspace{10mm}
\begin{center}
\textbf{ACKNOWLEDGEMENTS}
\end{center}
\vspace{10mm}

We thank B. Bennett for technical assistance and advice; E. Melamud and M. Clasquin for help with peak-picking; W. Lu and S. Kyin for mass spectrometry maintenance; C. DeCoste for FACS support; E. Smith, J. Shapiro, E. Anderson, I. Ehrenreich, and J. Bloom for advice. Funding provided by NIH grants R37 MH059520 to L.K., and GM071508 to the Lewis- Sigler Institute, and National Science Foundation CAREER Award MCB-0643859, Beckman Foundation and American Heart Association awards to J.D.R. L.K. is a James S. McDonnell Centennial Fellow and an Investigator of the Howard Hughes Medical Institute.

\vspace{10mm}
\begin{center}
\textbf{REFERENCES}
\end{center}
\vspace{10mm}



\bibliography{bpcite}

\end{document}

